content/guide/15-contributing-guide.md

```diff
@@ -0,0 +1,147 @@
+
+#### Adding our new fix!
+
+We have our failing test and now we want to make it pass.
+
+This where a good deal of work will take place. As a first time contributor digging around the codebase will be your new hobby. However, we have a leg up since we suspect it's a validation based bug. Immediately we start digging into `src/validate/` and see it's split up into `html`, `js`, and `utils`.
+
+![src/validate/ files](../../assets/images/svelte_src_validate_files.png)
+
+Since we're dealing with the code chunk `<div ref:foo-bar></div>` we can be confident in looking into the `html` first.
+
+![src/validate/html files](../../assets/images/svelte_src_validate_html.png).
+
+Great! Looks like we're getting closer to the source of the error. My first suspicion is that our fix will need to go in `validateComponent.ts` since we're validating on the component. (Spoiler: we're also going to need to add the code to `validateElement.ts`).
+
+In `validateComponent.ts` we find a conditional regarding an attribute's type being "Ref". 
+
+```js
+if (attribute.type === 'Ref') {
+  if (!refs.has(attribute.name)) refs.set(attribute.name, []);
+  refs.get(attribute.name).push(node);
+}
+```
+
+Looks like there's no check for determining valid names, so let's go ahead and add that. It just so happens that there's a utility function `isValidIdentifier` in `src/validate/utils/` that we can use to determine if our reference's name is valid. We can now find another place where this function is used to see if we can see how others have utilized this function.
+
+In `src/validate/js/propValidators/computed.ts` we see the function being used for what looks to be the same reason! 
+
+
+```js
+if (!isValidIdentifier(name)) {
+  const suggestion = name.replace(/[^_$a-z0-9]/ig, '_').replace(/^\d/, '_$&');
+  validator.error(computation.key, {
+    code: `invalid-computed-name`,
+    message: `Computed property name '${name}' is invalid — must be a valid identifier such as ${suggestion}`
+  });
+}
+```
+
+Now we can add a similar code block to our `validateComponent.ts` (and subsequently `validateElement.ts`).
+
+```js
+if (attribute.type === 'Ref') {
+  if (!isValidIdentifier(attribute.name)) {
+    const suggestion = attribute.name.replace(/[^_$a-z0-9]/ig, '_').replace(/^\d/, '_$&');
+    
+    validator.error(attribute, {
+      code: `invalid-reference-name`,
+      message: `Reference name '${attribute.name}' is invalid — must be a valid identifier such as ${suggestion}`
+    });	
+  } else {
+    if (!refs.has(attribute.name)) refs.set(attribute.name, []);
+    refs.get(attribute.name).push(node);
+  }
+}
+```
+
+Running `npm run test` again we can see we now have a passing test. Following standards, we can make our commits and then make a pull request for our new fix.
+
+All that's left to do is eagerly wait feedback and do it all over again!
```

Comment:
wait feedback -> wait for feedback

Reply:
I think I meant for it to be await feedback, thanks anyways!